packages/common/lib/types/Authorization.types.ts

```diff
@@ -1,14 +1,51 @@
-import { CredentialFormat } from '@sphereon/ssi-types';
+import { CommonCredentialOfferRequestPayload, CredentialOfferRequestWithBaseUrl } from './CredentialIssuance.types';
+import {
+  CredentialFormatEnum,
+  EndpointMetadata,
+  ErrorResponse,
+  IssuerCredentialDefinition,
+  IssuerCredentialSubject,
+  PRE_AUTH_CODE_LITERAL,
+} from './Generic.types';
+
+export interface CommonAuthorizationRequest {
+  response_type: ResponseType.AUTH_CODE;
+  client_id: string;
+  code_challenge: string;
+  code_challenge_method: CodeChallengeMethod;
+  redirect_uri: string;
+  scope?: string;
+  authorization_details?: CommonAuthorizationDetails[];
+  wallet_issuer?: string;
+  user_hint?: string;
+}
 
-import {CommonCredentialOfferRequestPayload, CredentialOfferRequestWithBaseUrl} from './CredentialIssuance.types';
-import { EndpointMetadata, ErrorResponse, PRE_AUTH_CODE_LITERAL } from './Generic.types';
+export interface AuthorizationRequestJwtVcJson extends CommonAuthorizationRequest {
+  authorization_details?: AuthorizationDetailsJwtVcJson[];
+}
+
+export interface AuthorizationRequestJwtVcJsonLdAndLdpVc extends CommonAuthorizationRequest {
+  authorization_details?: AuthorizationDetailsJwtVcJsonLdAndLdpVc[];
+}
 
-export interface AuthorizationDetails {
+export interface CommonAuthorizationDetails {
   type: 'openid_credential' | string;
+  format: CredentialFormatEnum;
   // If the Credential Issuer metadata contains an authorization_server parameter, the authorization detail's locations common data field MUST be set to the Credential Issuer Identifier value.
   locations?: string[];
-  format: CredentialFormat;
-  [s: string]: unknown;
+  [key: string]: unknown;
```

Comment:
This is probably one of the few cases any is better in case the values can also be objects. Reason is that you then can access subproperties. On an unknown JSC will throw errors

Reply:
changed to `any`